WordPress/Jetpack/Classes/System/JetpackWindowManager.swift

```diff
@@ -2,64 +2,74 @@ import Combine
             return
         }
 
-        guard FeatureFlag.contentMigration.enabled else {
-            showSignInUI()
+        // Show App UI if user is logged in
+        if AccountHelper.isLoggedIn {
+            self.showAppUI(for: blog)
             return
         }
 
-        self.migrationTracker.trackContentImportEligibility(eligible: shouldImportMigrationData)
-        shouldImportMigrationData ? importAndShowMigrationContent(blog) : showSignInUI()
+        // Show Sign In UI if previous conditions are false
+        self.showSignInUI()
     }
+}
 
-    func importAndShowMigrationContent(_ blog: Blog? = nil) {
-        self.migrationTracker.trackWordPressMigrationEligibility()
+// MARK: - Migration Related Properties and Methods
 
-        DataMigrator().importData() { [weak self] result in
-            guard let self else {
-                return
-            }
+extension JetpackWindowManager {
 
-            switch result {
-            case .success:
-                self.migrationTracker.trackContentImportSucceeded()
-                UserPersistentStoreFactory.instance().isJPContentImportComplete = true
-                NotificationCenter.default.post(name: .WPAccountDefaultWordPressComAccountChanged, object: self)
-                self.showMigrationUIIfNeeded(blog)
-                self.sendMigrationEmail()
-            case .failure(let error):
-                self.migrationTracker.trackContentImportFailed(reason: error.localizedDescription)
-                self.handleMigrationFailure(error)
-            }
+    @discardableResult
+    func startMigrationFlowIfNeeded(_ blog: Blog? = nil) -> Bool {
+        guard shouldStartMigrationFlow else {
+            return false
         }
+        self.importAndShowMigrationContent(blog)
+        return true
     }
 }
 
-// MARK: - Private Helpers
+// MARK: Private Helpers
 
 private extension JetpackWindowManager {
 
-    var shouldShowMigrationUI: Bool {
-        return FeatureFlag.contentMigration.enabled && AccountHelper.isLoggedIn
+    /// Checks whether the migration flow should start or not.
+    ///
+    /// This flag is `False` if a compatible WordPress app is not installed or the `contentMigration` feature flag is disabled.
+    ///
+    /// Otherwise, it is `True` when the following conditions are fulfilled:
+    ///
+    ///  1. User is not logged in and the migration is not complete.
+    ///  2. Or, User is logged in and the migration has started but still in progress. This scenario could happen if the migration flow starts but interrupted mid flow.
+    ///
+    var shouldStartMigrationFlow: Bool {
+        guard isCompatibleWordPressAppPresent && FeatureFlag.contentMigration.enabled else {
+            return false
+        }
+        let migrationState = UserPersistentStoreFactory.instance().jetpackContentMigrationState
+        let loggedIn = AccountHelper.isLoggedIn
+        return loggedIn ? migrationState == .inProgress : migrationState != .completed
     }
```

Comment:
I want to mention here that in the previous implementation, there was 2 flags:
- [shouldImportMigrationData](https://github.com/wordpress-mobile/WordPress-iOS/blob/4de5d26cb91d5458a39894ec13b09f815bc65edd/WordPress/Jetpack/Classes/System/JetpackWindowManager.swift#L11): This flag was used to check whether to start the migration flow or not. 
- [shouldShowMigrationUI](https://github.com/wordpress-mobile/WordPress-iOS/blob/4de5d26cb91d5458a39894ec13b09f815bc65edd/WordPress/Jetpack/Classes/System/JetpackWindowManager.swift#L59): This flag is called after the content import is successful, and it's used to check whether to show the Migration UI or not. I think this flag is unnecessary, because after the content import is successful, the Migration UI should appear.

In the new implementation, `shouldImportMigrationData` was refactored to `shouldStartMigrationFlow` and `shouldShowMigrationUI` was removed.